### ISSUE TITLE
Deprecate RegistrationBean methods that don't apply to ServletListenerRegistrationBean

### ISSUE THREAD

**Issue by xiaoshuang-lu** [spring-projects/spring-boot#11342]
spring boot version: 1.5.9.RELEASE

YetAnotherConfiguration
```java
    @Bean
    public ServletListenerRegistrationBean yetAnotherServletContextListener() {
        YetAnotherServletContextListener servletContextListener = new YetAnotherServletContextListener();

        ServletListenerRegistrationBean<ServletContextListener> servletListenerRegistrationBean =
            new ServletListenerRegistrationBean<ServletContextListener>();

        servletListenerRegistrationBean.setListener(servletContextListener);
        servletListenerRegistrationBean.addInitParameter("test", "test");
        servletListenerRegistrationBean.setOrder(0);

        return servletListenerRegistrationBean;
    }
```

YetAnotherServletContextListener
````java
    @Override
    public void contextInitialized(ServletContextEvent servletContextEvent) {
        // value is null?
        String value = servletContext.getInitParameter("test");
    }
````

**Comment by philwebb** [comment]
@xiaoshuang-lu It looks like we've made a mistake by having `ServletListenerRegistrationBean` extend `RegistrationBean`. The `addInitParameter` is designed to add init parameters to the `Servlet` or `Filter` registration. It doesn't change the context-wide initialization parameters.

There's not a good way to fix this in 1.5 without breaking back compatibility. Probably the best that we can do is deprecate those methods then fix the inheritance problem in 2.0.

**Comment by xiaoshuang-lu** [comment]
Thanks, @philwebb.
